builtin/logical/ad/path_config.go

```diff
@@ -0,0 +1,219 @@
+				Type:        framework.TypeBool,
+				Default:     false,
+				Description: "Skip LDAP server SSL Certificate verification - VERY insecure (optional)",
+			},
+
+			"starttls": {
+				Type:        framework.TypeBool,
+				Default:     true,
+				Description: "Issue a StartTLS command after establishing unencrypted connection (optional)",
+			},
+
+			"tls_min_version": {
+				Type:        framework.TypeString,
+				Default:     "tls12",
+				Description: "Minimum TLS version to use. Accepted values are 'tls10', 'tls11' or 'tls12'. Defaults to 'tls12'",
+			},
+
+			"tls_max_version": {
+				Type:        framework.TypeString,
+				Default:     "tls12",
+				Description: "Maximum TLS version to use. Accepted values are 'tls10', 'tls11' or 'tls12'. Defaults to 'tls12'",
+			},
+
+			"default_password_ttl": {
+				Type:        framework.TypeInt,
+				Default:     defaultPasswordTTLs,
+				Description: "In minutes, the default password time-to-live.",
+			},
+
+			"max_password_ttl": {
+				Type:        framework.TypeInt,
+				Default:     defaultPasswordTTLs,
+				Description: "In minutes, the maximum password time-to-live.",
+			},
+
+			"password_length": {
+				Type:        framework.TypeInt,
+				Default:     defaultPasswordLength,
+				Description: "The desired length of passwords that Vault generates.",
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.DeleteOperation: h.DeleteOperation,
+			logical.ReadOperation:   h.ReadOperation,
+			logical.UpdateOperation: h.UpdateOperation,
+		},
+	}
+}
+
+// Config is a convenience method for other operations to use in retrieving the current config.
+func (h *configHandler) Config(ctx context.Context, req *logical.Request) (*engineConfig, error) {
+
+	entry, err := req.Storage.Get(ctx, "config")
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return &engineConfig{}, nil
+	}
+
+	engineConf := &engineConfig{}
+	if err := entry.DecodeJSON(engineConf); err != nil {
+		return nil, err
+	}
+
+	return engineConf, nil
+}
+
+func (h *configHandler) DeleteOperation(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	if err := req.Storage.Delete(ctx, "config"); err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+func (h *configHandler) ReadOperation(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+
+	engineConf, err := h.Config(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &logical.Response{
+		Data: engineConf.Map(),
+	}
+	resp.AddWarning("Read access to this endpoint should be controlled via ACLs as it will return the configuration information as-is, including any passwords.")
+	return resp, nil
+}
+
+func (h *configHandler) UpdateOperation(ctx context.Context, req *logical.Request, fieldData *framework.FieldData) (*logical.Response, error) {
+
+	activeDirectoryConf, err := activedirectory.NewConfiguration(h.logger, fieldData)
```

Comment:
We might want to distinguish between a CreateOperation and an UpdateOperation. For that to work, we generally use `GetOk` on the field data. As of now, this overwrites the configuration entirely every time.

Reply:
Overwriting it every time is what I was going for because I figured it would be easier to do than having more PUT-like updates where they just change one field. I figured it wouldn't be too often it's updated so it might not be work the effort? Plus, other LDAP back ends just use write. However, if you feel there's a value to it, I can add it?